### ISSUE TITLE
espree Cannot read property 'type' of null

### ISSUE THREAD

**Issue by whitneyit** [eslint/espree#281]
The following `Promise.resolve(['a']).then(([,,,,]) => 1);` is reported as a `Parsing Error` however that is not correct.

It is the result of an acorn bug which was fixed in [`54fde50`](https://github.com/ternjs/acorn/commit/54fde50051100004faa0825c6ec7957ee9caf81d) and has yet to be [merged](https://github.com/ternjs/acorn/compare/3.2.0...master).

Cheers.

---

```js
"use strict";

const espree = require("espree");
const ast = espree.parse(
    "Promise.resolve(['a']).then(([,,,,]) => 1);",
    {ecmaVersion: 6}
);

console.log("espree", JSON.stringify(ast, null, 4));
```

```
C:\Users\james.whitney\Dev\node_modules\acorn\dist\acorn.js:1077
    if (isBinding && last.type === "RestElement" && last.argument.type !== "Identifier") this.unexpected(last.argument.start);
                         ^

TypeError: Cannot read property 'type' of null
    at Parser.pp.toAssignableList (C:\Users\james.whitney\Dev\node_modules\acorn\dist\acorn.js:1077:26)
    at Parser.pp.toAssignable (C:\Users\james.whitney\Dev\node_modules\acorn\dist\acorn.js:1030:14)
    at Parser.toAssignable (C:\Users\james.whitney\Dev\node_modules\espree\espree.js:297:37)
    at Parser.pp.toAssignableList (C:\Users\james.whitney\Dev\node_modules\acorn\dist\acorn.js:1081:19)
    at Parser.pp.parseArrowExpression (C:\Users\james.whitney\Dev\node_modules\acorn\dist\acorn.js:590:22)
    at Parser.pp.parseParenArrowList (C:\Users\james.whitney\Dev\node_modules\acorn\dist\acorn.js:418:15)
    at Parser.pp.parseParenAndDistinguishExpression (C:\Users\james.whitney\Dev\node_modules\acorn\dist\acorn.js:386:19)
    at Parser.pp.parseExprAtom (C:\Users\james.whitney\Dev\node_modules\acorn\dist\acorn.js:307:19)
    at Parser.pp.parseExprSubscripts (C:\Users\james.whitney\Dev\node_modules\acorn\dist\acorn.js:228:19)
    at Parser.pp.parseMaybeUnary (C:\Users\james.whitney\Dev\node_modules\acorn\dist\acorn.js:207:17)
```

**Comment by nzakas** [comment]
Thanks, I'll keep an eye out for when Acorn is updated.

**Comment by kaicataldo** [comment]
Looks like this is no longer an issue!
